interop-testing/src/test/java/io/grpc/testing/integration/TransportCompressionTest.java

```diff
@@ -0,0 +1,240 @@
+import io.grpc.testing.integration.Messages.Payload;
+import io.grpc.testing.integration.Messages.PayloadType;
+import io.grpc.testing.integration.Messages.SimpleRequest;
+import io.grpc.testing.integration.Messages.SimpleResponse;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.FilterInputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Tests that compression is turned on.
+ */
+@RunWith(JUnit4.class)
+public class TransportCompressionTest extends AbstractTransportTest {
+
+  private static int serverPort = TestUtils.pickUnusedPort();
+
+  private static final DecompressorRegistry decompressors = DecompressorRegistry.newEmptyInstance();
+  private static final CompressorRegistry compressors = CompressorRegistry.newEmptyInstance();
+
+  @Before
+  public void beforeTests() {
+    Fzip.INSTANCE.totalRead = 0;
+    Fzip.INSTANCE.totalWritten = 0;
+  }
+
+  @After
+  public void afterTests() {
+    // Just check that the compressor was involved.
+    assertTrue(Fzip.INSTANCE.anyRead || Fzip.INSTANCE.anyWritten);
+  }
+
+  /** Start server. */
+  @BeforeClass
+  public static void startServer() {
+    decompressors.register(Fzip.INSTANCE, true);
+    compressors.register(Fzip.INSTANCE);
+    startStaticServer(
+        ServerBuilder.forPort(serverPort)
+            .compressorRegistry(compressors)
+            .decompressorRegistry(decompressors),
+        new ServerInterceptor() {
+          @Override
+          public <ReqT, RespT> Listener<ReqT> interceptCall(MethodDescriptor<ReqT, RespT> method,
+              ServerCall<RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+            Listener<ReqT> listener = next.startCall(method, call, headers);
+            // TODO(carl-mastrangelo): check that encoding was set.
+            call.setMessageCompression(true);
+            return listener;
+          }});
+  }
+
+  /** Stop server. */
+  @AfterClass
+  public static void stopServer() {
+    stopStaticServer();
+  }
+
+  @Test
+  public void compresses() {
+    final SimpleRequest request = SimpleRequest.newBuilder()
+        .setResponseSize(314159)
+        .setResponseType(PayloadType.COMPRESSABLE)
+        .setPayload(Payload.newBuilder()
+            .setBody(ByteString.copyFrom(new byte[271828])))
+        .build();
+    final SimpleResponse goldenResponse = SimpleResponse.newBuilder()
+        .setPayload(Payload.newBuilder()
+            .setType(PayloadType.COMPRESSABLE)
+            .setBody(ByteString.copyFrom(new byte[314159])))
+        .build();
+
+    assertEquals(goldenResponse, blockingStub.unaryCall(request));
+    // Assert that compression took place
+    assertTrue(Fzip.INSTANCE.totalRead == Fzip.INSTANCE.totalWritten);
```

Comment:
Maybe move this line to a `@After` method, and remove this test method?
Also check `totalRead > 0`.